pandas/core/nanops.py

```diff
@@ -815,8 +816,9 @@ def _ensure_numeric(x):
         elif is_object_dtype(x):
             try:
                 x = x.astype(np.complex128)
-            except:
-                x = x.astype(np.float64)
+            except (TypeError, ValueError):
+                raise TypeError('Could not convert {value!s} to numeric'
+                                .format(value=x))
```

Comment:
I don't understand this change either. Seems like the original behavior was to try to cast to float. Why to not try it anymore and raise directly?